[Quant][docs] Replace qconfig_dict with QConfigMapping in docs

andrewor14 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #78533

Summary: https://github.com/pytorch/pytorch/pull/78452 replaced
qconfig_dict with QConfigMapping as the default API for prepare_fx,
prepare_qat_fx, and convert_fx. We should update the docs to reflect
this change as well.

Test Plan:
```
cd docs
make html
cd build/html
python -m server.http
```

Reviewers: jerryzh168, vkuzo

Subscribers: jerryzh168, vkuzo

andrewor14 committed:
[Quant][docs] Replace qconfig_dict with QConfigMapping in docs

Summary: https://github.com/pytorch/pytorch/pull/78452 replaced
qconfig_dict with QConfigMapping as the default API for prepare_fx,
prepare_qat_fx, and convert_fx. We should update the docs to reflect
this change as well.

Test Plan:
```
cd docs
make html
cd build/html
python -m server.http
```

Reviewers: jerryzh168, vkuzo

Subscribers: jerryzh168, vkuzo

[ghstack-poisoned]

vkuzo commented on docs/source/quantization.rst:
```diff
@@ -322,10 +322,11 @@ to do the following in addition:
 
 API Example::
 
+  from torch.ao.quantization import QConfigMapping
```
nit: `torch.quantization` for now to stay consistent with the rest of the document?

we plan to communicate the `ao` part once all the refactors are landed

andrewor14 committed:
Update on "[Quant][docs] Replace qconfig_dict with QConfigMapping in docs"


Summary: https://github.com/pytorch/pytorch/pull/78452 replaced
qconfig_dict with QConfigMapping as the default API for prepare_fx,
prepare_qat_fx, and convert_fx. We should update the docs to reflect
this change as well.

Test Plan:
```
cd docs
make html
cd build/html
python -m server.http
```

Reviewers: jerryzh168, vkuzo

Subscribers: jerryzh168, vkuzo

[ghstack-poisoned]

andrewor14 commented:
@pytorchbot merge this please
